packages/hint/src/lib/cli/analyze.ts

```diff
@@ -104,46 +104,55 @@ const pruneUserConfig = (userConfig: UserConfig) => {
     };
 };
 
-/** Ask user if he wants to activate the telemetry or not. */
-const askForTelemetryConfirmation = async (userConfig: UserConfig) => {
-    if (appInsights.isConfigured()) {
-        return;
-    }
-
-    if (isCI) {
-        if (!appInsights.isConfigured()) {
-            showCITelemetryMessage();
-        }
+const askForTelemetryConfirmation = async () => {
+    showTelemetryMessage();
 
-        return;
-    }
+    const message = `Do you want to opt-in?`;
 
-    const alreadyRun: boolean = configStore.get(configStoreKey);
+    debug(`Prompting telemetry permission.`);
 
-    if (!alreadyRun) { /* This is the first time, don't ask anything. */
-        configStore.set(configStoreKey, true);
+    const telemetryEnabled = await askQuestion(message);
 
-        return;
+    if (telemetryEnabled) {
+        appInsights.enable();
+    } else {
+        appInsights.disable();
     }
 
-    showTelemetryMessage();
-
-    const message: string = `Do you want to opt-in?`;
-
-    debug(`Prompting telemetry permission.`);
+    return telemetryEnabled;
+};
 
-    const confirm: boolean = await askQuestion(message);
+/** Ask user if he wants to activate the telemetry or not. */
+const sendTelemetryIfEnabled = async (userConfig: UserConfig) => {
+    const telemetryConfigured = appInsights.isConfigured();
+    let telemetryEnabled = appInsights.isEnabled();
+    const alreadyRun: boolean = configStore.get(alreadyRunKey);
 
-    if (confirm) {
-        appInsights.enable();
+    if (!alreadyRun) {
+        configStore.set(alreadyRunKey, true);
+    }
 
-        appInsights.trackEvent('SecondRun');
-        appInsights.trackEvent('analyze', pruneUserConfig(userConfig));
+    if (!telemetryConfigured) {
+        if (isCI) {
+            showCITelemetryMessage();
+            telemetryEnabled = false;
+        } else if (alreadyRun) {
+            /* Only prompt the user about opt-intelemetry if they have run webhint before. */
+            telemetryEnabled = await askForTelemetryConfirmation();
+        }
+    }
 
+    if (!telemetryEnabled) {
         return;
     }
 
-    appInsights.disable();
+    appInsights.trackEvent('analyze', {
+        ...{
+            ci: isCI,
+            previouslyRun: alreadyRun
+        },
```

Comment:
Similar to my earlier feedback, there's no need to create an intermediate object that we just throw away:

```ts
    appInsights.trackEvent('analyze', {
        ci: isCI,
        previouslyRun: alreadyRun,
        ...pruneUserConfig(userConfig)
    });
```

Reply:
Done